### PR TITLE
[Keyboard] Add dz60rgb_ansi v2_1

### PR DESCRIPTION
**Pull request by daskygit**
## Description

Adds a v2_1 like the other boards, I did need to add `#define LAYER_STATE_8BIT` to keep it under size with my avr-gcc version (10.2.0). Hopefully there's not too many other lufa-ms updated boards.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by daskygit** [comment]
Just noticed #14376 is trying to add support under the version `dz60rgb_ansi/v2_1_chrisgve` although judging from the suggested changes, maybe drashna missed it wasn't just a keymap change?

File: keyboards/dztech/dz60rgb_ansi/v2_1/config.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_DIGITAL_RAIN
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_ADDR_2 0b1010000 // this is here for compliancy reasons.
+#    define DRIVER_COUNT 2
+#    define DRIVER_1_LED_TOTAL 61
+#    define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+#endif
+
+#define LAYER_STATE_8BIT
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb_ansi/v2_1/config.h:65]:
```suggestion
```
Configurator supports up to 16 layers so this should not be set at the keyboard level.

File: keyboards/dztech/dz60rgb_ansi/v2_1/config.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#    define DISABLE_RGB_MATRIX_SPLASH
+#    define DISABLE_RGB_MATRIX_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_SPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_DIGITAL_RAIN
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_ADDR_2 0b1010000 // this is here for compliancy reasons.
+#    define DRIVER_COUNT 2
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb_ansi/v2_1/config.h:60]:
```suggestion
#    define DRIVER_COUNT 1
```
*Fairly* certain this was fixed.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/dztech/dz60rgb_ansi/v2_1/config.h (hunk 2 above)
Comment (daskygit) [reply keyboards/dztech/dz60rgb_ansi/v2_1/config.h:60]: Compiles fine with this change but no hardware to to test with.

**Comment by daskygit** [comment]
Enabled LTO so it'll fit when compiled with version other than 8.4.

File: keyboards/dztech/dz60rgb_ansi/v2_1/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#    define DISABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#    define DISABLE_RGB_MATRIX_SPLASH
+#    define DISABLE_RGB_MATRIX_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_SPLASH
+#    define DISABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#    define DISABLE_RGB_MATRIX_DIGITAL_RAIN
+#    define DRIVER_ADDR_1 0b1010000
+#    define DRIVER_COUNT 1
+#    define DRIVER_1_LED_TOTAL 61
+#    define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb_ansi/v2_1/config.h:61]:
```suggestion
#    define DRIVER_LED_TOTAL 61
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/dztech/dz60rgb_ansi/v2_1/config.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
+#pragma once
```
Comment (drashna) [inline keyboards/dztech/dz60rgb_ansi/v2_1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by daskygit** [comment]
Done, assuming I can give copyright to QMK?

**Review by drashna — APPROVED** [review]

**Pull request merged**